Adds CLI explain command

phil-scott-78 opened the pull request:
Wanted a command to break down how Spectre views the command model. This outputs all the relevant settings in a tree. You can get a short explanation for all the commands, or a detailed explanation for all commands (there is a flag to go vice versa on the detailed view if you want)

```
app cli explain
app cli explain myappcommand
app cli explain -d
```

I honestly only included the output that I think is helpful for my use cases, so I'm sure I missed something. Feel free to suggest new data, better, formatting or what-have-you!

patriksvensson commented on src/Spectre.Console/Cli/Internal/Commands/ExplainCommand.cs:
```diff
@@ -0,0 +1,244 @@
+{
+    [Description("Displays diagnostics about CLI configurations")]
+    [SuppressMessage("Performance", "CA1812: Avoid uninstantiated internal classes")]
+    internal class ExplainCommand : Command<ExplainCommand.Settings>
```
I prefer that we seal classes that are not designed for (further) inheritance.

patriksvensson commented on src/Spectre.Console/Cli/Internal/Commands/ExplainCommand.cs:
```diff
@@ -0,0 +1,244 @@
+
+        public ExplainCommand(IConfiguration configuration, CommandModel commandModel)
+        {
+            _commandModel = commandModel;
```
```suggestion
            _commandModel = commandModel ?? throw new ArgumentNullException(nameof(commandModel));
```

patriksvensson commented on src/Spectre.Console/Cli/Internal/Commands/ExplainCommand.cs:
```diff
@@ -0,0 +1,244 @@
+        public sealed class Settings : CommandSettings
+        {
+            [CommandArgument(0, "[command]")]
+            public string[]? Command { get; set; } = null;
```
Rename this to `Commands` or similar?

patriksvensson reviewed (changes requested):
Some nitpicking things, but otherwise this looks good to me!

phil-scott-78 commented:
Pushed up a change with the requests featured. I also rebased on main to get the nullable constructor bugfixes from earlier and adjusted the `Settings` to use a constructor with get only properties.

phil-scott-78 commented:
ok, now the changes are there. My git client was sitting there asking me to confirm a git force for 9 hours.

patriksvensson reviewed (approved):
LGTM!
